### ISSUE TITLE
Show indeterminate status for sharing permissions

### ISSUE THREAD

**Issue by MorrisJobke** [nextcloud/server#8371]
We should use the `indeterminate` checkbox state for the "can edit" checkbox in the sharing settings if not all of "can create", "can change" and "can delete" is checked, otherwise it would be confusing for the user when "can edit" is shown but actually less permissions are given - especially with #7363 being used.

![bildschirmfoto 2018-02-15 um 10 54 41](https://user-images.githubusercontent.com/245432/36251197-97cc0d16-1240-11e8-9e4e-c8f5147758c3.png)

See also https://stackoverflow.com/a/6269487/520507

Found while looking into #7363 

cc @nextcloud/javascript @nextcloud/designers

**Comment by dasisdormax** [comment]
Hey. If that's a good beginner issue, I'd like to look into it if you don't mind.

**Comment by MorrisJobke** [comment]
@dasisdormax Sure go ahead :) If you need help let us know and we can guide you. Maybe also join the #nextcloud-dev channel on IRC on freenode. Maybe @skjnldsv @danxuliu or @juliushaertl already can give some tips :) 🎉

**Comment by MorrisJobke** [comment]
This looks like a good starting point:

https://github.com/nextcloud/server/blob/05bcdf937b9976de560192a3b06c21e5b6ea7774/core/js/sharedialogshareelistview.js#L32-L35

**Comment by dasisdormax** [comment]
Yep, found it already. Thanks ^^

**Comment by dasisdormax** [comment]
Just noticed that the HTML attribute `indeterminate="indeterminate"` doesn't set the indeterminate property of the checkbox automatically the way `checked="checked"` does. I think we can't set it through the template but have to do it in the render function.

**Comment by skjnldsv** [comment]
@dasisdormax no indeed, you're supposed to change the state with js
``` js
var checkbox = document.getElementById("some-checkbox");
checkbox.indeterminate = true;
```

**Comment by pixelipo** [comment]
https://css-tricks.com/indeterminate-checkboxes/

**Comment by dasisdormax** [comment]
Alright, thanks for your help so far!

I think I got it working, so what would be the best way to submit the code?

**Comment by skjnldsv** [comment]
@dasisdormax create a new pull request from your repository :)

**Comment by pixelipo** [comment]
You can ask to join the team and make a branch and PR directly on https://github.com/nextcloud/server/

**Comment by MorrisJobke** [comment]
Fixed by #8390
